packages/ms-vc-api-issuer/src/agent/MsVcApiIssuer.ts

```diff
@@ -0,0 +1,90 @@
+import { IAgentPlugin } from '@veramo/core'
+import { IIssueRequest, IIssueRequestResponse, IMsVcApiIssuer, IRequiredContext } from '../types/IMsVcApiIssuer'
+import { v4 as uuidv4 } from 'uuid'
+import { ClientCredentialAuthenticator} from '@sphereon/ms-authenticator'
+/**
+ * {@inheritDoc IMsVcApiIssuer}
+ */
+export class MsVcApiIssuer implements IAgentPlugin {
+  readonly methods: IMsVcApiIssuer = {
+    issuanceRequestMsVc: this.issuanceRequestMsVc.bind(this)
+  }
+
+  private msIdentityHostName = ''
+
+  /** {@inheritDoc IMsVcApiIssuer.issuanceRequestMsVc} */
+  private async issuanceRequestMsVc(issuanceInfo: IIssueRequest, context: IRequiredContext): Promise<IIssueRequestResponse> {
+    var accessToken = await ClientCredentialAuthenticator(issuanceInfo.authenticationInfo);
+    await fetch(`https://login.microsoftonline.com/${issuanceInfo.authenticationInfo.azTenantId}/v2.0/.well-known/openid-configuration`, { method: 'GET' })
+      .then(res => res.json())
+      .then(async (resp) => {
+        this.msIdentityHostName = "https://beta.did.msidentity.com/v1.0/";
+        if (resp.tenant_region_scope == "EU") {
+          this.msIdentityHostName = "https://beta.eu.did.msidentity.com/v1.0/";
+        }
+        // Check that the Credential Manifest URL is in the same tenant Region and throw an error if it's not
+        if (!issuanceInfo.authenticationInfo.credentialManifestUrl.startsWith(this.msIdentityHostName)) {
+          throw new Error(`Error in config file. CredentialManifest URL configured for wrong tenant region. Should start with:${this.msIdentityHostName}`);
+        }
+      });
+
+    // Config Request and App Config File should be a parameter to this function
+    if (!issuanceInfo.authenticationInfo.azTenantId) {
+      throw new Error('The config.json file is missing.')
+    }
+
+    issuanceInfo.issuanceConfig.registration.clientName = "Sphereon Node.js SDK API Issuer";
+
+    // modify the callback method to make it easier to debug
+    // with tools like ngrok since the URI changes all the time
+    // this way you don't need to modify the callback URL in the payload every time
+    // ngrok changes the URI
+    issuanceInfo.issuanceConfig.callback.url = `https://6270-2a02-a458-e71a-1-68b4-31d2-b44f-12b.eu.ngrok.io/api/issuer/issuance-request-callback`;
```

Comment:
Why are we doing this in the method body?

Reply:
Moved to the test code